bxserum/provider/grpc.go

```diff
@@ -7,15 +7,24 @@ import (
 	"github.com/bloXroute-Labs/serum-client-go/bxserum/transaction"
 	pb "github.com/bloXroute-Labs/serum-client-go/proto"
 	"github.com/gagliardetto/solana-go"
+	log "github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
+	"sync"
+	"time"
 )
 
 type GRPCClient struct {
 	pb.UnimplementedApiServer
 
+	mutex *sync.RWMutex
+
 	apiClient  pb.ApiClient
 	privateKey *solana.PrivateKey
+
+	recentBlockHash               *pb.GetRecentBlockHashResponse
+	recentBlockHashTime           *time.Time
```

Comment:
any reason this needs to be a pointer? Seems to result in some unnecessary dereferencing

Reply:
just to be sure that we're not returning empty values, but yes, empty string is not valid blockHash :)

Reply:
I mean time, for hash, which contains mutex it's necessary